sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosChangeFeedItemsDataSource.scala

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import org.apache.spark.sql.connector.catalog.{Table, TableProvider}
+import org.apache.spark.sql.connector.expressions.Transform
+import org.apache.spark.sql.sources.DataSourceRegister
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+import java.util
+
+// scalastyle:off underscore.import
+import scala.collection.JavaConverters._
+// scalastyle:on underscore.import
+
+private class CosmosChangeFeedItemsDataSource
+  extends DataSourceRegister
+    with TableProvider
```

Comment:
Motivation: we will support different capabilities - Items won't support  MICRO_BATCH_READ or CONTINUOUS_READ - just BATCH_READ (and the former will drive the design even for BATCH_READ in change feed to make sure BATCH_READ reuses the code used for streaming. Changefeed won't support any writes - so the common denominator is BATCH_READ (implemented differently).
I don't think it will be helpful to use common base classes for both. But maybe we can take another look at it when we have feature-complete implementations for both - at that point it will be more obvious which part is common code and whether it is worth refactoring it so that we reduce code-duplication?

Reply:
Cassandra doesn't support MICRO_BATCH_READ or CONTINOUS_READ - just STREAMING_WRITE (so similar to our "Items" Data Source)

Reply:
Dynamo DB doesn't support streaming at all

Reply:
Kafka implementation is here - https://github.com/apache/spark/blob/master/external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

Reply:
But that one is really just similar to ChangeFeed - doesn't have query with push-down capabilities like the "Items" DataSource in Cosmos

Reply:
Thanks @FabianMeiswinkel  for the explanation. skeleton looks good to me, maybe after feature complete once we know more we can decide whether we can share code or not.